apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx

```diff
@@ -54,23 +97,37 @@ export function ExternalDappLoader({
             url: input,
           })
         }
-      })
-      .catch((error) => {
-        if (!isRequestRelevant) return
-
-        console.error(error)
-        setManifestError('Can not fetch the manifest.json')
-      })
-      .finally(() => {
-        if (!isRequestRelevant) return
+      } catch (error) {
+        console.error('Hook dapp loading error:', error)
 
+        if (error.message?.includes('JSON')) {
```

Comment:
I don't get this check. Maybe what you want to check is the mimetype of the response?

Reply:
Added a MIME type check now that gets the content-type header from the response, verifies that it exists and includes application/json.
It will show a descriptive error if the content type is wrong.

This will help catch cases where the URL points to an HTML page or other non-JSON content before we try to parse it. 

The original error handling for error instanceof Error is still useful as a final fallback (it returns the raw error message)